### PR TITLE
CIV-8454 Generate service request

### PR DESCRIPTION
**Pull request by paul-pearson**
### JIRA link ###

https://tools.hmcts.net/jira/browse/CIV-8454

### Change description ###

Fire event to trigger service request creation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -49,7 +49,8 @@ src/main/routes/features/claimantResponse/partPaymentReceivedController.ts, \
 src/main/routes/features/claimantResponse/ccj/ccjConfirmationController.ts, \
 src/main/routes/features/directionsQuestionnaire/supportRequiredController.ts, \
 src/main/routes/features/directionsQuestionnaire/witnesses/otherWitnessesController.ts, \
-src/main/assets/js/fileValidator.js
+src/main/assets/js/fileValidator.js, \
+src/main/routes/features/claim/totalAmountController.ts
```
Comment (laarmada) [inline sonar-project.properties:53]: which is the issue to skip sonnar? totalAmountController

File: src/main/services/features/claim/amount/claimFeesService.ts [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 
 const {Logger} = require('@hmcts/nodejs-logging');
 const logger = Logger.getLogger('claimantPhoneAsService');
-export const saveClaimFee = async (claimantId: string, claimFee: number) => {
+export const saveClaimFee = async (claimantId: string, claimFeeData: ClaimFeeData) => {
   try{
     const claim = await getCaseDataFromStore(claimantId);
     console.log(claimantId);
-    claim.claimFee = {calculatedAmountInPence: convertToPence(claimFee)};
+    claim.claimFee = {
```
Comment (laarmada) [inline src/main/services/features/claim/amount/claimFeesService.ts:10]: only a question which journey is this claimFee?

**Review by dharmendrak — APPROVED** [review]

File: src/main/app/client/civilServiceClient.ts [hunk 3]
```diff
@@ -155,11 +155,16 @@ export class CivilServiceClient {
...
+    const claimFeeData = await this.getClaimFeeData(amount, req);
+    return convertToPoundsFilter(claimFeeData?.calculatedAmountInPence.toString());
+  }
+
+  async getClaimFeeData(amount: number, req: AppRequest): Promise<ClaimFeeData> {
     const config = this.getConfig(req);
     try {
       const response: AxiosResponse<object> = await this.client.get(`${CIVIL_SERVICE_CLAIM_AMOUNT_URL}/${amount}`, config);
       const claimFeeResponse: ClaimFeeData = response.data;
-      return convertToPoundsFilter(claimFeeResponse?.calculatedAmountInPence.toString());
```
Comment (laarmada) [inline src/main/app/client/civilServiceClient.ts:162]: really good to see that removed once is adding in civil-service 
image

**Review by laarmada — COMMENTED** [review]
ok

**Review by jeswanth-hmcts — APPROVED** [review]

File: sonar-project.properties (hunk 1 above)
Comment (paul-pearson) [reply sonar-project.properties:53]: All the claim tests are disabled in jest here: https://github.com/hmcts/civil-citizen-ui/blob/2275b1f95c92c268ad714e675f53d018015d5d1b/jest.config.js#L5
So this means I can't get test coverage for my changes in totalAmountController.ts.
I think @portizdev has been working on fixing the broken tests which will allow us to re-enable all these tests and remove sonar exclusions.

File: src/main/services/features/claim/amount/claimFeesService.ts (hunk 2 above)
Comment (paul-pearson) [reply src/main/services/features/claim/amount/claimFeesService.ts:10]: It is the same journey - claim creation.  On screen /claim/total we calculate the fee from civil service request.  Previously we were only saving the fee calculatedAmountInPence.  After this change we will be saving the additional fields so we can use them in service request creation.

File: sonar-project.properties (hunk 1 above)
Comment (laarmada) [reply sonar-project.properties:53]: yes but please don't add more files in the exclusions... we need to figure out the reason and try to fix it. That works is included in the story points estimations.

File: src/main/services/features/claim/amount/claimFeesService.ts (hunk 2 above)
Comment (laarmada) [reply src/main/services/features/claim/amount/claimFeesService.ts:10]: we should stop to add isolated vbles in claim directly we need to create hierarchy in order to know in which journey is used that data.... else we will have the same problem as CCD

**Review by laarmada — APPROVED** [review]

**Pull request merged**